src/utils/index.ts

```
@@ -29,3 +29,22 @@ export const nodeHexToID = (h) => {
 export const shortenNodeId = (nodeId: string) => {
   return nodeId.slice(0, 12).concat('...').concat(nodeId.slice(-6))
 }
+
+export const nodeIDRemovePrefix = (pk: string): string => {
+  if (!pk.startsWith('NodeID-')) {
+    throw new Error("Error: nodeID must start with 'NodeID-'")
+  }
+  try {
+    const prefix = 'NodeID-'
+    const startIndex = pk.indexOf(prefix)
+
+    if (startIndex === -1) {
+      throw new Error('Error: nodeID is incorrect length')
+    }
+
+    // Extracting the NodeID
+    return pk.substring(startIndex + prefix.length)
+  } catch (e) {
+    throw new Error("Error: nodeID must start with 'NodeID-'")
```

`'Error: nodeID is incorrect length'` will never be thrown

---

Also no need to re-throw the error inside the catch block, just `console.warn(e)` or handle it however you would want to handle this type of error.